src/modeler/AutoRest.Swagger.Tests/SwaggerModelerValidationTests.cs

```diff
@@ -41,20 +42,25 @@ internal static void AssertOnlyValidationMessage(this IEnumerable<ValidationMess
     [Collection("Validation Tests")]
     public partial class SwaggerModelerValidationTests
     {
-        private IEnumerable<ValidationMessage> ValidateSwagger(string input)
+        private IEnumerable<ValidationMessage> ValidateSwagger(string input, ServiceDefinitionDocumentType openapiDocType = ServiceDefinitionDocumentType.ARM)
         {
             using (NewContext)
             {
-                new Settings
+                var messages = new List<LogMessage>();
+                var validator = new RecursiveObjectValidator(PropertyNameResolver.JsonName);
+                var serviceDefinition = SwaggerParser.Parse(input, File.ReadAllText(input));
+
+                var metaData = new ServiceDefinitionMetadata
                 {
-                    CodeGenerator = "None",
-                    Namespace = "Test",
-                    Input = input
+                    OpenApiDocumentType = openapiDocType,
+                    MergeState = ServiceDefinitionMergeState.After
                 };
-                var modeler = new SwaggerModeler();
-                var messages = new List<LogMessage>();
+
                 Logger.Instance.AddListener(new SignalingLogListener(Category.Info, messages.Add));
```

Comment:
Simply return `validator.GetValidationExceptions(new Uri(input), serviceDefinition, metaData)` instead of setting up a listener only to collect back the messages.